src/Widgets/SharePopover.vala

```diff
@@ -31,135 +31,76 @@ public class SharePopover : Gtk.Popover {
-        copy_link_button.get_style_context ().add_class (Gtk.STYLE_CLASS_FLAT);
-
-        var size_group = new Gtk.SizeGroup (Gtk.SizeGroupMode.BOTH);
-        size_group.add_widget (email_button);
-        size_group.add_widget (copy_link_button);
-
+        var email_button = make_share_button (
+            _("Email"), "mailto:?body=%s %s", "internet-mail");
+        var copy_link_button = make_copy_link_button ();
+        
         var service_grid = new Gtk.Grid ();
         service_grid.margin = 6;
         service_grid.add (email_button);
-        service_grid.add (facebook_button);
-        service_grid.add (google_button);
-        service_grid.add (twitter_button);
-        service_grid.add (reddit_button);
-        service_grid.add (tumblr_button);
-        service_grid.add (telegram_button);
-
+        service_grid.add (make_share_button (
+            _("Facebook"), "https://www.facebook.com/sharer/sharer.php?u=%s","online-account-facebook"));
+        service_grid.add (make_share_button (
+            _("Twitter"), "http://twitter.com/home/?status=%s %s", "online-account-twitter"));
+        service_grid.add (make_share_button (
+            _("reddit"), "http://www.reddit.com/submit?title=%s&url=%s", "online-account-reddit"));
+        service_grid.add (make_share_button (
+            _("Tumblr"), "https://www.tumblr.com/share/link?url=%s", "online-account-tumblr"));
+        service_grid.add (make_share_button (
+            _("Telegram"), "https://t.me/share/url?url=%s", "online-account-telegram"));
+        
         var system_grid = new Gtk.Grid ();
         system_grid.margin = 6;
         system_grid.add (copy_link_button);
 
+        var size_group = new Gtk.SizeGroup (Gtk.SizeGroupMode.BOTH);
+        size_group.add_widget (copy_link_button);
+        size_group.add_widget (email_button);
+
         var grid = new Gtk.Grid ();
         grid.orientation = Gtk.Orientation.VERTICAL;
         grid.add (service_grid);
         grid.add (new Gtk.Separator (Gtk.Orientation.HORIZONTAL));
         grid.add (system_grid);
         grid.show_all ();
-
         add (grid);
+    }
 
-        copy_link_button.clicked.connect (() => {
-            var clipboard = Gtk.Clipboard.get_for_display (get_display (), Gdk.SELECTION_CLIPBOARD);
-            clipboard.set_text (uri, -1);
-
-            link_copied ();
-
-            hide ();
-        });
-
-        email_button.clicked.connect (() => {
-            try {
-                AppInfo.launch_default_for_uri ("mailto:?body=%s %s".printf (body, uri), null);
-            } catch (Error e) {
-                warning ("%s", e.message);
-            }
-            hide ();
-        });
-
-        facebook_button.clicked.connect (() => {
+    Gtk.Button make_share_button (string name, string uri_template, string icon_name) {
+        var share_button = new Gtk.Button.from_icon_name (icon_name, Gtk.IconSize.DND);
+        share_button.tooltip_text = name;
```

Comment:
In general, it's probably best to name these variables the same as the properties they correspond to for clarity. So instead of `name` this should be `tooltip_text`